### PR TITLE
r? ianbicking Fixes #478 Resize the panel when redisplaying it, depending on whether we are editing the current shot or not

### LINKED ISSUE #478: When you redisplay a page that has an active shot, and you have edited it, it is not resized
It starts over with the small size:

screen shot 2015-06-09 at 10 50 41 am

The large size should be sticky.  Maybe a property on the ShotContext.

### PR DESCRIPTION
**Pull request by fzzzy**

### CODE REVIEW

File: addon/lib/main.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ const { ToggleButton } = require('sdk/ui/button/toggle');
 const panels = require("sdk/panel");
 const { watchFunction, watchWorker } = require("./errors");
 
+let isEditing = new Map();
```
Comment (ianb) [inline addon/lib/main.js]: Why can't this just be a property on the ShotContext?

File: addon/lib/main.js [hunk 2]
```diff
@@ -102,13 +104,18 @@ const PanelContext = {
     shotContext.isShowing();
     shootButton.checked = true;
     this.updateShot(this._activeContext, this._activeContext.shot.asJson());
+    if (isEditing.get(this._activeContext.shot.id)) {
```
Comment (ianb) [inline addon/lib/main.js]: I.e., here it would be `if (this._activeContext.isEditing)`

File: addon/lib/main.js [hunk 3]
```diff
@@ -150,19 +157,19 @@ const PanelContext = {
         backend: shotContext.shot.backend,
         id: shotContext.shot.id,
         shot: shotContext.shot.asJson(),
-        activeClipName: shotContext.activeClipName
+        activeClipName: shotContext.activeClipName,
+        isEditing: isEditing.get(shotContext.shot.id)
       }
```
Comment (ianb) [inline addon/lib/main.js]: And then right here, do a shootPanel.resize() based on shotContext.isEditing

File: addon/lib/shooter.js [hunk 4]
```diff
@@ -302,8 +302,8 @@ const ShotContext = Class({
     openLink: function (link) {
       tabs.open(link);
     },
-    setSize: function(size) {
-      this.panelContext.setSize(size);
+    setEditing: function(editing) {
```
Comment (ianb) [inline addon/lib/shooter.js]: And this is `this.isEditing = true` then `this.updateShot()`

File: addon/lib/main.js (hunk 1 above)
Comment (fzzzy) [inline addon/lib/main.js]: It can, I just didn't think of it

**Comment by ianb** [comment]
I think the logic will be simpler moving stuff into ShotContext

**Comment by fzzzy** [comment]
PR updated to store isEditing on the shot context.

**Comment by ianb** [comment]
Merging, but something to test:

1. Start PS on tab A
2. Start PS on tab B
3. Switch to tab A and hit "editing"
4. Switch back and forth between two tabs

**Pull request merged**